src/edwards.rs

```diff
@@ -432,32 +432,41 @@ impl<'a, 'b> Mul<&'b Scalar> for &'a ExtendedPoint {
     /// For scalar multiplication of a basepoint,
     /// `EdwardsBasepointTable` is approximately 4x faster.
     fn mul(self, scalar: &'b Scalar) -> ExtendedPoint {
-        // Construct a lookup table of [P,2P,3P,4P,5P,6P,7P,8P]
-        let lookup_table = LookupTable::<ProjectiveNielsPoint>::from(self);
+        // If we built with AVX2, use the AVX2 backend.
+        #[cfg(all(feature="nightly", all(feature="avx2_backend", target_feature="avx2")))] {
```

Comment:
What is the reasoning behind having the `#[cfg(…)]` inside the function rather than just have different functions with the same interface that are chosen at compile time?

Reply:
Right now the backend selection is done totally statically, choosing the AVX2 backend if `yolocrypto` is on and the target supports AVX2. In the future we might want to do dynamic dispatch, so that the library could have two codepaths, and in that case the selection has to be done inside the highest-level user-facing API.

It would probably be cleaner to keep the non-vectorized implementation in another, seperate function, but it's a bit awkward because that implementation works with some backends (the `u64` and `u32` ones) but not this one.